src/editline.c

```diff
@@ -140,26 +156,17 @@ ec_editline_set_prompt(struct ec_editline *editline, const char *prompt)
 			return -1;
 	}
 
+	if (el_set(editline->el, EL_PROMPT, prompt_cb))
+		goto fail;
```

Comment:
I have read the commit message several times but I fail to understand how that works.

What is the purpose of that callback?